docs/fp-SDK-guide.md

```diff
@@ -0,0 +1,414 @@
+
+## 1. Introduction
+
+The Finality Provider SDK enables Bitcoin Supercharged Networks (BSNs) to 
+integrate with Babylon's Bitcoin-secured finality infrastructure as a flexible 
+library. It provides finality guarantees for consumer chain blocks through 
+EOTS (Extractable One-Time Signatures) and BTC staking.
+
+The SDK offers a modular, interface-based architecture that allows teams to 
+customise components for their specific blockchain implementations while 
+maintaining compatibility with the core finality provider functionality.
+
+The SDK includes two reference implementations:
+- **Cosmos chains** - Uses CosmWasm smart contracts for finality 
+  coordination ([implementation](../bsn/cosmos/), [config](../bsn/cosmos/config/))
+- **Rollup chains** - Uses Ethereum-compatible smart contracts for finality 
+  coordination ([implementation](../bsn/rollup/), [config](../bsn/rollup/config/))
+
+Both implementations demonstrate the same extensible interface pattern, 
+enabling BSNs to create custom implementations for any consumer chain type 
+by implementing [interfaces](../clientcontroller/api/interface.go) 
+rather than modifying core logic.
+
+## 2. Core constructor
+
+The BSN SDK uses `service.NewFinalityProviderApp()` as the main entry point. It 
+supports two usage modes:
+
+### 2.1. Default Usage
+Use built-in implementations with constructors. These create all
+required interfaces internally with default implementations:
+
+```go
+// Cosmos chains - Uses CosmWasm client, Cosmos SDK queries
+fpApp, err := cosmosservice.NewCosmosBSNFinalityProviderAppFromConfig(
+    cfg, db, logger)
+
+// Rollup chains - Uses Ethereum RPC client, JSON-RPC queries
+fpApp, err := rollupservice.NewRollupBSNFinalityProviderAppFromConfig(
+    cfg, db, logger)
+```
+
+Both internally call [`service.NewFinalityProviderApp()`](../finality-provider/service/app.go)
+with pre-configured implementations.
+
+### 2.2. Custom Usage
+Implement your own interfaces and call the core constructor directly. This
+allows custom consumer chain types beyond Cosmos/Rollup:
+
+```go
+fpApp, err := service.NewFinalityProviderApp(
+    config,              // *fpcfg.Config 
+    babylonController,   // ccapi.BabylonController - Can be customised
+    consumerController,  // ccapi.ConsumerController - Can be customised
+    eotsManager,         // eotsmanager.EOTSManager - Can be customised
+    blockPoller,         // types.BlockPoller - Can be customised
+    randomnessCommitter, // types.RandomnessCommitter - Can be customised
+    heightDeterminer,    // types.HeightDeterminer - Can be customised
+    finalitySubmitter,   // types.FinalitySignatureSubmitter - Can be customised
+    metrics,             // *metrics.FpMetrics - Metrics collection
+    db,                  // kvdb.Backend - Database for state persistence
+    logger,              // *zap.Logger - Structured logging
+)
+```
+You can find the above example at [`app.go`](../finality-provider/service/app.go)
```

Comment:
How about we add notice showing example how we did it for rollup or cosmos? It would be useful so others can see what they need to do. https://github.com/babylonlabs-io/finality-provider/blob/sam/fp-sdk/bsn/rollup/service/app.go#L17